scene/resources/animation.cpp

```diff
@@ -5594,16 +5594,20 @@ Variant Animation::array_to_string(const Variant p_value) {
 }
 
 Variant Animation::add_variant(const Variant &a, const Variant &b) {
-	if (a.get_type() != b.get_type() && !a.is_array()) {
-		return a;
+	if (a.get_type() != b.get_type()) {
+		if (a.is_num() && b.is_num()) {
+			return add_variant(a.operator double(), b.operator double());
```

Comment:
```suggestion
			return add_variant(cast_to_blendwise(a), cast_to_blendwise(b));
```
Same for the others.

Reply:
Done. 👍